src/hooks/useSubStep/index.ts

```diff
@@ -1,63 +1,88 @@
 import type {ComponentType} from 'react';
-import {useCallback, useRef, useState} from 'react';
+import {useCallback, useMemo, useRef, useState} from 'react';
 import type {SubStepProps, UseSubStep} from './types';
 
+function calculateLastIndex(bodyContentLength: number, skipSteps: number[] = []) {
+    let lastIndex = bodyContentLength - 1;
+    while (skipSteps.includes(lastIndex)) {
+        lastIndex -= 1;
+    }
+
+    return lastIndex;
+}
+
 /**
  * This hook ensures uniform handling of components across different screens, enabling seamless integration and navigation through sub steps of the VBBA flow.
  * @param bodyContent - array of components to display in particular step
  * @param onFinished - callback triggered after finish last step
  * @param startFrom - initial index for bodyContent array
  * @param onNextSubStep - callback triggered after finish each step
+ * @param skipSteps - array of indexes to skip
  */
-export default function useSubStep<TProps extends SubStepProps>({bodyContent, onFinished, startFrom = 0, onNextSubStep = () => {}}: UseSubStep<TProps>) {
+export default function useSubStep<TProps extends SubStepProps>({bodyContent, onFinished, startFrom = 0, skipSteps = [], onNextSubStep = () => {}}: UseSubStep<TProps>) {
     const [screenIndex, setScreenIndex] = useState(startFrom);
     const isEditing = useRef(false);
 
+    if (bodyContent.length === skipSteps.length) {
+        throw new Error('All steps are skipped');
+    }
+
+    const lastScreenIndex = useMemo(() => calculateLastIndex(bodyContent.length, skipSteps), [bodyContent.length, skipSteps]);
+
```

Comment:
Hmm, could you help to share problems which you get? I think this way will be cleaner and simpler

Reply:
@shubham1206agra 


> I think we should filter out skipSteps from bodyContent and

If we filter out skipSteps from bodyContent, I think we don't need to change anything to logic to navigate to the next step and the previous step.

Could you please detail this point?

>  it will be hard to determine the next or previous step. And the indexes will be jumbled if I want to do a onMove step.

Reply:
@shubham1206agra Could you bring your thoughts to this thread?